src/dotnet/APIView/APIView/AssemblyAPIV.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace APIView
+{
+    /// <summary>
+    /// Class representing a C# assembly. Each assembly has a name and global namespace, 
+    /// which may or may not contain further types.
+    /// 
+    /// Assembly is an immutable, thread-safe type.
+    /// </summary>
+    public class AssemblyAPIV
+    {
+        public string Name { get; }
+        public NamespaceAPIV GlobalNamespace { get; }
+
+        /// <summary>
+        /// Construct a new Assembly instance, represented by the provided symbol.
+        /// </summary>
+        /// <param name="symbol">The symbol representing the assembly.</param>
+        public AssemblyAPIV(IAssemblySymbol symbol)
+        {
+            this.Name = symbol.Name;
+            this.GlobalNamespace = new NamespaceAPIV(symbol.GlobalNamespace);
+        }
+
+        public static List<AssemblyAPIV> AssembliesFromFile(string dllPath)
+        {
+            var reference = MetadataReference.CreateFromFile(dllPath);
+            var compilation = CSharpCompilation.Create(null).AddReferences(reference);
+            compilation = compilation.AddReferences(MetadataReference.CreateFromFile(typeof(object).Assembly.Location));
```

Comment:
You don't have the TRUSTED_PLATFORM_ASSEMBLY fix here so I doubt that this code path would work correctly.

Reply:
That's true - I'm writing up the fix to include this right now. Wanted to get things up in a PR and then tackle that after.